sixtyfps_compiler/generator/rust.rs

```diff
@@ -1598,12 +1598,26 @@ fn compile_expression(expr: &Expression, component: &Rc<Component>) -> TokenStre
             let (padding, spacing) = generate_layout_padding_and_spacing(&layout.geometry, *o, component);
             let cells = grid_layout_cell_data(layout, *o, component);
             let size = layout_geometry_size(&layout.geometry.rect, *o, component);
-            quote!(solve_grid_layout(&GridLayoutData{
-                size: #size,
-                spacing: #spacing,
-                padding: #padding,
-                cells: Slice::from_slice(&#cells),
-            }))
+            if let (Some(button_roles), Orientation::Horizontal) = (&layout.dialog_button_roles, *o) {
+                let role = button_roles.iter().map(|x| format_ident!("{}", x));
+                quote!({
+                    let mut cells = #cells;
+                    reorder_dialog_button_layout(&mut cells, &[ #(DialogButtonRole::#role),* ]);
+                    solve_grid_layout(&GridLayoutData{
+                        size: #size,
+                        spacing: #spacing,
+                        padding: #padding,
+                        cells: Slice::from_slice(&cells),
+                    })
+                })
+            } else {
+                quote!(solve_grid_layout(&GridLayoutData{
+                    size: #size,
+                    spacing: #spacing,
+                    padding: #padding,
+                    cells: Slice::from_slice(&#cells),
+                }))
```

Comment:
Would be nice to get rid of the else since it's the same solve_grid_layout call, but perhaps that would make the code harder to read again as it requires concatenating the token stream. Just a thought - feel free to ignore :)